template/web/src/actions/get-collection-metadata.ts

```diff
@@ -0,0 +1,125 @@
+// Minimal ABI with contractURI view function
+import { Abi, Address } from 'abitype';
+import { PublicClient } from 'viem';
+import { ipfsToHTTP } from '@/utils/ipfs';
+
+const AbiFunctionContractURI = [
+  {
+    inputs: [],
+    name: 'contractURI',
+    outputs: [{ internalType: 'string', name: '', type: 'string' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+] as const;
+
+// Minimal ABI with uri view function
+const AbiFunctionWithUri = [
+  {
+    inputs: [{ internalType: 'uint256', name: 'id', type: 'uint256' }],
+    name: 'uri',
+    outputs: [{ internalType: 'string', name: '', type: 'string' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+] as const;
+
+/**
+ * There is some differences in URI standards between ERC721 and 1155, handle those in this component.
+ */
+const uriFunctionType = {
+  uri: 'uri',
+  contractURI: 'contractURI',
+} as const;
+
+const getIsAbiWithContractURIViewFunction = (abi: Abi) =>
```

Comment:
Sorry, can we write this code in a more verbose way.

It's a bit hard to follow when it's all written in one solo line of command.

Same in other places in this file.